par_loop_generator.cpp

```diff
@@ -422,19 +425,21 @@ void emit_loop_body(loop_code* i_code, char* body_func_name) {
         add_buf_to_code(i_code, tmp_buf);
         align_line(i_code);
         sprintf(tmp_buf, "idx[%d] = layer_id;\n", 3 + 3*i_code->n_logical_loops);
-        add_buf_to_code(i_code, tmp_buf);        
+        add_buf_to_code(i_code, tmp_buf);
       } else {
         align_line(i_code);
         sprintf(tmp_buf, "idx[%d] = 0;\n", i + i_code->n_logical_loops);
         add_buf_to_code(i_code, tmp_buf);  
       }
     }
   } else {
-    align_line(i_code);
-    sprintf(tmp_buf, "idx[%d] = tid;\n", 0 + 3*i_code->n_logical_loops);
-    add_buf_to_code(i_code, tmp_buf); 
+    if (i_code->logical_loop_partype[i] != (int)THREADED_LOOP_NO_PARALLEL) {
+      align_line(i_code);
+      sprintf(tmp_buf, "idx[%d] = tid;\n", 0 + 3*i_code->n_logical_loops);
```

Comment:
Extra if fixes the problem when idx[] = tid statement is emitted without tid defined ( = when no emit_parallel_loop is called before)

Reply:
Looks good to me!